metricq_sink_nsca/check.py

```diff
@@ -164,6 +163,23 @@ def _has_value_checks(self) -> bool:
     def _has_timeout_checks(self) -> bool:
         return self._timeout_checks is not None
 
+    async def _heartbeat(self):
+        while True:
+            await sleep(self._resend_interval.s)
+            now = Timestamp.now()
+            (state, message) = self.format_overall_state()
+
+            self._last_report_triggered_time = now
+
+            logger.debug("Sending heartbeat")
+            # fire and forget
+            create_task(
+                self._send_result_callback(
```

Comment:
I'm wondering, whether this needs an asyncio.Lock to protect the `state`.

Reply:
Eh... that could be the case :thinking: will have a look.

`Check` is starting to look a lot like :spaghetti:, I'd like to rewrite this to use a producer-consumer queue. Then `ReporterSink` and `Check` are not coupled as tightly and sending asynchronous check results like this does not cause as much headache.